src/Raven.Server/Documents/Sharding/Handlers/Processors/Streaming/ShardedStreamingHandlerProcessorForGetStreamQuery.cs

```diff
@@ -0,0 +1,258 @@
+    {
+        public ShardedStreamingHandlerProcessorForGetStreamQuery([NotNull] ShardedDatabaseRequestHandler requestHandler, HttpMethod method) : base(requestHandler, method)
+        {
+        }
+        
+        protected override RequestTimeTracker GetTimeTracker()
+        {
+            return new RequestTimeTracker(HttpContext, Logger, RequestHandler.DatabaseContext.NotificationCenter, RequestHandler.DatabaseContext.Configuration, "StreamQuery", doPerformanceHintIfTooLong: false);
+        }
+
+        protected override async ValueTask<BlittableJsonReaderObject> GetDocumentDataAsync(TransactionOperationContext context, string fromDocument)
+        {
+            var shard = RequestHandler.DatabaseContext.GetShardNumber(context, fromDocument);
+            
+            var docs = await RequestHandler.ShardExecutor.ExecuteSingleShardAsync(context,
+                new GetDocumentsCommand(new [] { fromDocument }, includes: null, metadataOnly: false), shard);
+            return (BlittableJsonReaderObject)docs.Results[0];
+        }
+
+        protected override IDisposable AllocateContext(out TransactionOperationContext context)
+        {
+            return ContextPool.AllocateOperationContext(out context);
+        }
+
+        protected override QueryMetadataCache GetQueryMetadataCache()
+        {
+            return RequestHandler.DatabaseContext.QueryMetadataCache;
+        }
+
+        protected override IStreamQueryResultWriter<BlittableJsonReaderObject> GetBlittableQueryResultWriter(string format, bool isDebug, JsonOperationContext context, HttpResponse response, Stream responseBodyStream, bool fromSharded,
+            string[] propertiesArray, string fileNamePrefix = null)
+        {
+            if (IsCsvFormat(format))
+            {
+                //does not write query stats to stream
+                return new StreamCsvBlittableQueryResultWriter(response, responseBodyStream, propertiesArray, fileNamePrefix);
+            }
+            
+            if (isDebug)
+            {
+                ThrowUnsupportedException($"You have selected \"{format}\" file format, which is not supported.");
+            }
+            
+            return new StreamBlittableDocumentQueryResultWriter(responseBodyStream, context);
+        }
+
+        private async ValueTask<(IEnumerator<BlittableJsonReaderObject>, StreamQueryStatistics)> ExecuteQueryAsync(TransactionOperationContext context, IndexQueryServerSide query, string debug, bool ignoreLimit, OperationCancelToken token)
+        {
+            var queryProcessor = new ShardedQueryStreamProcessor(context, RequestHandler, query);
+            if (queryProcessor.IsMapReduce())
+                throw new NotSupportedInShardingException("MapReduce is not supported in sharded streaming queries");
+
+            //TODO stav: should have this?
```

Comment:
we do not support includes while streaming

Reply:
now throwing in regular streaming EP if Includes or Load